File: katt1/friends/include/python3/grader.py
#!/usr/bin/python3
import friends
from sys import stdin

N, L, Q = [int(x) for x in stdin.readline().split()]
P = [int(x) for x in stdin.readline().split()]
friends.init(N, L, P)
for i in range(Q):
    t = [int(x) for x in stdin.readline().split()]
    if t[0] == 1:
        print(friends.score())
    elif t[0] == 0:
        friends.jump(t[1], t[2])
    else:
        assert False

File: katt1/friends3/include/cpp/grader.cpp
#include "friends.h"
#include <cstdio>
#include <cassert>

int main() {
  int N, L, Q;
  scanf("%d%d%d", &N, &L, &Q);
  int P[N];
  for (int i = 0; i < N; ++i) scanf("%d", P + i);
  init(N, L, P);
  for (int i = 0; i < Q; ++i) {
    int type;
    scanf("%d", &type);
    if (type == 0) {
      int A, B;
      scanf("%d%d", &A, &B);
      jump(A, B);
    }
    else if (type == 1) printf("%lld\n", score());
    else assert(false && "Invalid query type");
  }
}
